pkg/kubelet/kubelet.go

```diff
@@ -929,34 +929,107 @@ type Kubelet struct {
 	runtimeCache    kubecontainer.RuntimeCache
 	kubeClient      clientset.Interface
 	heartbeatClient clientset.Interface
-	rootDirectory   string
+	// mirrorPodClient is used to create and delete mirror pods in the API for static
+	// pods.
+	mirrorPodClient kubepod.MirrorClient
+
+	rootDirectory string
 
 	lastObservedNodeAddressesMux sync.RWMutex
 	lastObservedNodeAddresses    []v1.NodeAddress
 
 	// onRepeatedHeartbeatFailure is called when a heartbeat operation fails more than once. optional.
 	onRepeatedHeartbeatFailure func()
 
-	// podWorkers handle syncing Pods in response to events.
+	// podManager stores the desired set of admitted pods and mirror pods that the kubelet should be
+	// running. The actual set of running pods is stored on the podWorkers. The manager is populated
+	// by the kubelet config loops and may be consulted by other components that need to see the set
+	// of desired pods. Note that not all desired pods are running, and not all running pods are in
+	// the podManager - for instance, force deleting a pod from the apiserver will remove it from
+	// the podManager, but the pod may still be terminating and tracked by the podWorkers. Components
```

Comment:
should we note here that modifying/deleting a static pod manifest will also have similar effect as force deleting pod from API server here?